src/library/parsercsv.cpp

```diff
@@ -41,29 +41,26 @@ QList<QString> ParserCsv::parseAllLocations(const QString& playlistFile) {
 
         QList<QList<QString>> tokens = tokenize(bytes, ',');
 
-        // detect Location column
-        int locationColumnIndex = -1;
+        const auto detect_location_column = [&](const auto& tokens_list, auto predicate) {
+            const auto it = std::find_if(std::begin(tokens_list), std::end(tokens_list), predicate);
+            return (it != std::end(tokens_list))
+                    ? std::distance(std::begin(tokens_list), it)
+                    : std::optional<size_t>{std::nullopt};
```

Comment:
```suggestion
        const auto detect_location_column = [&](const auto& tokens_list, auto predicate) -> std::optional<std::size_t> {
            const auto it = std::find_if(std::begin(tokens_list), std::end(tokens_list), predicate);
            return (it != std::end(tokens_list))
                    ? std::distance(std::begin(tokens_list), it)
                    : std::nullopt;
```

Reply:
Hi @Swiftb0y, sadly compilers cannot work with your suggestion. :crying_cat_face: 
The compilers need to guess the datatype to be assigned to the `return` keyword by the point of the  `?`-keyword in the ternary operator. 
`std::nullopt` is too vague of a (partially-specialised) datatype for the compiler to "strong-type"; the trailing-return datatype-hint does nothing in this regard sadly.

---
How about this code as a middle-ground?
```cpp
const auto detect_location_column = [&](const auto& tokens_list, auto predicate) {
    const auto it = std::find_if(std::begin(tokens_list), std::end(tokens_list), predicate);
    return (it != std::end(tokens_list))
            ? std::distance(std::begin(tokens_list), it)
            : std::optional<size_t>{};
};
```

Reply:
How about this? I primarily just care about the trailing return type for doc purposes. Also in C++ its preferred to use the prefixed `std::size_t` instead of C's `size_t` AFAIK.
```suggestion
        const auto detect_location_column = [&](const auto& tokens_list, auto predicate) -> std::optional<std::size_t>{
            const auto it = std::find_if(std::begin(tokens_list), std::end(tokens_list), predicate);
            return (it != std::end(tokens_list))
                    ? std::distance(std::begin(tokens_list), it)
                    : std::optional<std::size_t>{};
```